src/lsp/main.ts

```diff
@@ -350,10 +365,85 @@ async function stopClient() {
   }
 }
 
-async function startClientCommand() {
+enum FolderType {
+  VSCODE = 0,
+  FROM_FS_FILE = 1,
+  FROM_UNTITLED_FILE = 2,
+  FROM_NO_CLOJURE_FILE = 3,
+}
+
+type FallbackFolder = {
+  uri: vscode.Uri;
+  type: FolderType;
+};
+
+/**
+ * Figures out a ”best fit” rootUri for use when starting the clojure-lsp
+ * server. See: https://github.com/BetterThanTomorrow/calva/issues/1664
+ *
+ * @returns the ”best fit” rootUri or `undefined` if there is a VS Code folder open
+ */
+async function getFallbackFolder(): Promise<FallbackFolder> {
+  // If VS Code has a folder open, a rootUri is not needed
+  // we signal that with `undefined`
+  if (vscode.workspace.workspaceFolders) {
+    return {
+      uri: undefined,
+      type: FolderType.VSCODE,
+    };
+  }
+
+  const activeEditor = vscode.window.activeTextEditor;
+  let clojureFilePath: string;
+  let folderType: FolderType;
+  if (activeEditor && activeEditor.document?.languageId === 'clojure') {
+    folderType = activeEditor.document.isUntitled
+      ? FolderType.FROM_UNTITLED_FILE
+      : FolderType.FROM_FS_FILE;
+    if (!activeEditor.document.isUntitled) {
+      clojureFilePath = activeEditor.document.uri.fsPath;
+    }
+  } else {
+    for (const document of vscode.workspace.textDocuments) {
+      if (document.languageId === 'clojure') {
+        folderType = document.isUntitled ? FolderType.FROM_UNTITLED_FILE : FolderType.FROM_FS_FILE;
+        if (!document.isUntitled) {
+          clojureFilePath = document.uri.fsPath;
+        }
+      }
+    }
+  }
+
+  if (!clojureFilePath) {
+    if (!folderType) {
+      folderType = FolderType.FROM_NO_CLOJURE_FILE;
+    }
+    clojureFilePath = path.join(util.calvaTmpDir(), 'clojure-lsp-fallback-root', 'Untitled');
```

Comment:
Have you tried calling it untitled.clj? That might make it work in lsp

Reply:
I'm not naming the file here. I'm creating a path on which I can do `dirname`. Maybe needs a clarifying comment...